yt/frontends/gadget/fields.py

```diff
@@ -122,6 +128,22 @@ def _h_density(field, data):
 
         return ["H"] + metal_names[:-1]
 
+    def _check_whitelist_species_fields(self, ptype):
+        species_names = []
+        species_nums = []
+        for field in self.ds.field_list:
+            if field[0] == ptype:
+                if field[1].endswith(("_fraction", "_density")):
+                    if field[1] in sph_whitelist_fields:
+                        symbol, _, _ = field[1].partition("_")
+                        species_names.append(symbol)
+                        species_nums.append(periodic_table[symbol].num)
+        if len(species_names) > 0:
+            ret = list(sort_together([species_nums, species_names])[1])
+        else:
+            ret = []
+        return ret
```

Comment:
Now that I get what's done, here's a refactor suggestion. It's flatter, and doesn't require the exotic (but quite nice !) `sort_together` function
```suggestion
        species_names = []
        for field in self.ds.field_list:
            if (
                field[0] == ptype
                and field[1].endswith(("_fraction", "_density"))
                and field[1] in sph_whitelist_fields
            ):
                symbol, _, _ = field[1].partition("_")
                species_names.append(symbol)
        return sorted(specie_names, key=lambda symbol: periodic_table[symbol].num)
```